File: backend/api/app/domain/use_cases/post/delete.py
from core.use_cases.use_case import UseCase
from app.domain.repositories.post import BaseRepository
from core.common.equatable import Equatable
from core.common.either import Either
from core.errors.failure import Failure
from app.domain.entities.post import Post, PostEntity

class Params(Equatable):
    def __init__(self, post_id: str) -> None:
        self.post_id = post_id

class DeletePost(UseCase[PostEntity]):
    def __init__(self, repository: BaseRepository):
        self.repository = repository
    
    async def __call__(self, params: Params) -> Either[Failure, PostEntity]:
        return await self.repository.delete_post(params.post_id)

File: mobile/lib/features/architect/domains/use_cases/post/delete.dart
import '../../../../../core/errors/failure.dart';
import '../../entities/post.dart';
import '../../repositories/post.dart';
import 'package:dartz/dartz.dart';
import 'package:equatable/equatable.dart';

import '../../../../../core/use_cases/usecase.dart';

class DeletePost extends UseCase<Post, Params> {
  final PostRepository repository;

  DeletePost(this.repository);

  @override
  Future<Either<Failure, Post>> call(Params params) async {
    return await repository.delete(params.postId);
  }
}

class Params extends Equatable {
  final String postId;

  const Params({required this.postId});

  @override
  List<Object?> get props => [postId];
}
